### PR TITLE
Fixes case where a read for all channels fails and reports empty value.

### LINKED ISSUE #3044: Incorrect Asset status report from cloud call
**Describe the bug**
When a read for all the channels is requested by the cloud, if an error happens, for example the driver is not able to reach the target PLC, the catcher error is not reported, resulting in an empty list of channels:

```
[
  {
    "name": "modbus_asset_1",
    "channels": [
      {
        "name": "Channel-1",
        "timestamp": 1598004632308,
        "value": "4",
        "type": "INTEGER"
      },
      {
        "name": "Channel-2",
        "timestamp": 1598004632309,
        "value": "6",
        "type": "INTEGER"
      },
      {
        "name": "Channel-3",
        "timestamp": 1598004632310,
        "error": "Unknown error"
      }
    ]
  },
  { "name": "modbus_asset_2", "channels": [] },
  {
    "name": "modbus_asset_3",
    "channels": [
      {
        "name": "Channel-1",
        "timestamp": 1598004632348,
        "value": "4",
        "type": "INTEGER"
      }
    ]
  }
]
```

Asset 1 and 3 correctly report even if there is an error in a specific channel.
For asset 2, that has connection issues, what is reported is an empty list

**To Reproduce**
1. Define an asset that has a driver, for example Modbus, with a wrong target IP configured. It will return usually, also via local ESF REST call a connection error 
2. Using the cloud platform request a read for all the channels of the given asset
3. See error

**Expected behavior**
The result reported to the cloud platform should state that an error happened.

**Screenshots**
N/A

**Target Environment (please complete the following information):**
 - Board [e.g. Raspberry Pi 3]
 - OS version: [e.g. provide the result of `uname -a`]
 - Additional info: 

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by MMaiero**
Signed-off-by: Maiero <matteo.maiero@eurotech.com>

### CODE REVIEW

**Review by nicolatimeus — APPROVED** [review]

**Comment by MMaiero** [comment]
Closes #3044

**Comment by MMaiero** [comment]
@mstankovic Can you please validate this PR?

File: kura/org.eclipse.kura.asset.cloudlet.provider/src/main/java/org/eclipse/kura/internal/asset/cloudlet/AssetCloudlet.java [hunk 1]
```diff
@@ -22,12 +22,15 @@
 import java.util.Map.Entry;
 import java.util.Optional;
 import java.util.Set;
+import java.util.stream.Collectors;
```
Comment (LeoNerdoG) [inline kura/org.eclipse.kura.asset.cloudlet.provider/src/main/java/org/eclipse/kura/internal/asset/cloudlet/AssetCloudlet.java:25]: Please change header to "2020".

**Review by LeoNerdoG — CHANGES_REQUESTED** [review]
hey @MMaiero, please update the header in one of the files.

File: kura/org.eclipse.kura.asset.cloudlet.provider/src/main/java/org/eclipse/kura/internal/asset/cloudlet/AssetCloudlet.java (hunk 1 above)
Comment (MMaiero) [reply kura/org.eclipse.kura.asset.cloudlet.provider/src/main/java/org/eclipse/kura/internal/asset/cloudlet/AssetCloudlet.java:25]: Done

**Review by LeoNerdoG — APPROVED** [review]

**Pull request merged**